src/tests/users/updatePassword.test.js

```diff
@@ -0,0 +1,183 @@
+      status: 400,
+      body: {
+        success: false,
+        message: 'Invalid request',
+        errors: {
+          newPassword: ['New password must be at least 7 character mix of capital, small letters with numbers'],
+        },
+      },
+    },
+  },
+  {
+    title: 'should fail if new password does not contain a number',
+    request: context => ({
+      method: 'put',
+      path: `/api/users/${context.ward._id}/password`,
+      body: { formerPassword: 'Password123', newPassword: 'passworD' },
+      headers: {
+        'req-token': context.ward.authToken,
+      },
+    }),
+    response: {
+      status: 400,
+      body: {
+        success: false,
+        message: 'Invalid request',
+        errors: {
+          newPassword: ['New password must be at least 7 character mix of capital, small letters with numbers'],
+        },
+      },
+    },
+  },
+  {
+    title: 'should fail if Former password is incorrect',
+    request: context => ({
+      method: 'put',
+      path: `/api/users/${context.ward._id}/password`,
+      body: { formerPassword: 'Passwor3', newPassword: 'passworD223' },
+      headers: {
+        'req-token': context.ward.authToken,
+      },
+    }),
+    response: {
+      status: 400,
+      body: {
+        success: false,
+        message: 'Former password is not correct',
+      },
+    },
+  },
+  {
+    title: 'should fail if the user is not found',
+    request: context => ({
+      method: 'put',
+      path: '/api/users/5db95971c9da2412401b1804/password',
+      body: { formerPassword: context.ward.stringPass, newPassword: 'passworD223' },
+      headers: {
+        'req-token': context.ward.authToken,
+      },
+    }),
+    response: {
+      status: 404,
+      body: {
+        success: false,
+        message: 'User does not exist',
+      },
+    },
+  },
+  {
+    title: 'should fail if former password or new password is not a string',
+    request: context => ({
+      method: 'put',
+      path: `/api/users/${context.ward._id}/password`,
+      body: { formerPassword: 55289099, newPassword: 55555582 },
+      headers: {
+        'req-token': context.ward.authToken,
+      },
+    }),
+    response: {
+      status: 400,
+      body: {
+        success: false,
+        message: 'Invalid request',
+        errors: {
+          formerPassword: ['Former password is a required string'],
+          newPassword: ['New password is a required string'],
+        },
+      },
+    },
+  },
+  {
+    title: 'should fail if req-token is not in the header',
+    request: context => ({
+      method: 'put',
+      path: `/api/users/${context.ward._id}/password`,
+      body: { formerPassword: context.ward.stringPass, newPassword: 'passworD223' },
+      headers: {
+        Auth: context.testGlobals[HOSPITAL_ADMIN_USER].authToken,
+      },
+    }),
+    response: {
+      status: 401,
+      body: {
+        success: false,
+        message: 'Unable to authenticate token',
+      },
+    },
+  },
+];
```

Comment:
Use the confirmAccessLevelRestriction generic test case to confirm that only a ward user can update his/her account.

Reply:
`router.put('/users/:userId/password', authMiddleware, controllers.allUser.updatePassword);`
access level restriction was not added to the route since all user types (hospital admin, ward and nurse users) can access the route and update their respective passwords. 
The only check in place is found at the service level as shown in the snippet below
```
  const user = await db.users.getUser({ _id: userId }, userType);
  if (!user) {
    log.debug('The user to update does not exist');
    const error = new Error('User does not exist');
    error.httpStatusCode = 404;
    throw error;
  }
```
The check verifies the user exits.

Reply:
Makes sense!